ui/src/logs/api/index.ts

```diff
@@ -0,0 +1,23 @@
+import {proxy} from 'src/utils/queryUrlGenerator'
+import {Namespace} from 'src/types'
+
+export const executeQueryAsync = async (
+  proxyLink: string,
+  namespace: Namespace,
+  query: string
+) => {
+  try {
+    const {data} = await proxy({
+      source: proxyLink,
+      db: namespace.database,
+      rp: namespace.retentionPolicy,
+      query,
+      tempVars: [],
+      resolution: null,
+    })
+
+    return data
+  } catch (error) {
+    throw error
```

Comment:
could you add a dispatch for an error notification here?

Reply:
I don't think that we should dispatch from an api call, probably better to await and catch in the action that calls it and then dispatch an error message from there.

Reply:
ah yes, my mistake -- good call